discord/ext/commands/core.py

```diff
@@ -255,8 +257,16 @@ def __init__(self, func, **kwargs):
         # bandaid for the fact that sometimes parent can be the bot instance
         parent = kwargs.get('parent')
         self.parent = parent if isinstance(parent, _BaseCommand) else None
-        self._before_invoke = None
-        self._after_invoke = None
+
+        try:
+            self.before_invoke(func.__before_invoke__)
```

Comment:
```suggestion
            before_invoke = func.__before_invoke__
```

I like to keep one attribute access per exception. You can work with it inside `else:`